src/base/bittorrent/torrenthandle.h

```diff
@@ -330,6 +330,13 @@ namespace BitTorrent
         void addUrlSeeds(const QVector<QUrl> &urlSeeds);
         void removeUrlSeeds(const QVector<QUrl> &urlSeeds);
         bool connectPeer(const PeerAddress &peerAddress);
+        void setMaxConnections(int limit) const;
+        void setMaxUploads(int limit) const;
+
+        void moveQueuePositionUp() const;
```

Comment:
These 4 methods are incorrect at least as they're implemented currently. Each movement of torrent in the queue should be followed by queue adjustment (we need to adjust positions of "hidden" torrents that preload their metadata) so TorrentHandle cannot manage this action itself.

Reply:
OK, will revert.
Also what about catching exceptions? I think those are redundant.